lib/spack/llnl/util/cpu/detect.py

```diff
@@ -80,7 +80,7 @@ def check_output(args):
         return subprocess.run(
             args, check=True, stdout=subprocess.PIPE).stdout  # nopyqver
     else:
-        return subprocess.check_output(args)  # nopyqver
+        return subprocess.check_output(args).decode('utf-8')  # nopyqver
```

Comment:
I guess I missed this.  We should actually make this call `Popen` directly, as `check_output` is not actually in Python 2.6.  I'm curious whether this line is tested -- guess we'll find out from codecov. (FYI @alalazo)

Reply:
@tgamblin We check whether we are on Python 2.6 in the if-statement above.

Reply:
This line is hit on Darwin only, meaning we test it only in Python 2.7. Is it worth adding a new macos builder testing Python 3.7 ?